irc/client.go

```diff
@@ -1026,6 +1027,9 @@ func (client *Client) SetNames(username, realname string, fromIdent bool) error
 
 	if client.realname == "" {
 		client.realname = realname
+		if client.alwaysOn {
+			client.markDirty(IncludeRealname)
```

Comment:
This needs to happen outside the lock as well (you can use a defer for this)

Reply:
From discussion: there's actually no need to mark dirty here, since this is never called for registered clients.